processing/src/main/java/org/apache/druid/segment/UnnestDimensionCursor.java

```diff
@@ -181,7 +181,10 @@ public void inspectRuntimeShape(RuntimeShapeInspector inspector)
           @Override
           public Object getObject()
           {
-            if (indexedIntsForCurrentRow == null) {
+            if (dimSelector.getObject() == null) {
```

Comment:
Why did we switch away from using the stored reference?  `getObject()` can do work, work we've already done, doing the same work multiple times is bad.

It looks to me like you just needed to switch it to be 

```
if (indexedIntsForCurrentRow == null || indexedIntsForCurrentRow.size() == 0)
```

And then, you could perhaps make it even simpler if you check the size once when setting `indexedIntsForCurrentRow` and set to `null` when size is 0.

Reply:
Moving to stored reference and setting the `indexedIntsForCurrentRow` to null at the time of assignment if there are no elements